packages/@aws-cdk/aws-backup/lib/plan.ts

```diff
@@ -124,6 +133,7 @@ export class BackupPlan extends Resource implements IBackupPlan {
 
     const plan = new CfnBackupPlan(this, 'Resource', {
       backupPlan: {
+        ...(props.windowsVss ? { advancedBackupSettings: [{ backupOptions: { WindowsVSS: 'enabled' }, resourceType: 'EC2' }] } : {}),
```

Comment:
I don't like this line, there is too much going on. Especially because `advancedBackupSettings` includes other options other than `windowsVss`, and we don't want to have to restructure the code every time we want to add a new option. Instead, try this structure:

```ts
const plan = new CfnBackupPlan(this, 'Resource', {
  backupPlan: {
    advancedBackupSettings: this.advancedBackupSettings(props),
    ...
}

private advancedBackupSettings(props: CfnBackupPlanProps) {
  if (!props.windowsVss) {
    return undefined;
  }
  return [{
    backupOptions: {
      WindowsVSS: 'enabled',
    },
    resourceType: 'EC2',
  }];
}